src/Maestro/FeedCleaner/FeedCleaner.cs

```diff
@@ -0,0 +1,369 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Maestro.AzureDevOps;
+using Maestro.Contracts;
+using Maestro.Data;
+using Maestro.Data.Models;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.ServiceFabric.ServiceHost;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Internal;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace FeedCleaner
+{
+
+    /// <summary>
+    ///     An instance of this class is created for each service replica by the Service Fabric runtime.
+    /// </summary>
+    public sealed class FeedCleaner : IFeedCleaner, IServiceImplementation
+    {
+
+        private readonly string MaestroManagedFeedPattern = @"darc-(?<type>(int|pub))-(?<repository>.+?)-(?<sha>[A-Fa-f0-9]{7,40})-?(?<subversion>\d*)";
```

Comment:
This (or something like it) already exists in the GitFileManager. can we pull those two out into a common location (or use the DarcLib version)

Reply:
I'll pull them both out into a common place in DarcLib, They're not quite the same as the ones in GitFileManager care about the whole feed URL including the nuget/v3/index.json part, but they are so similar that I think it makes sense to have them in the same place.

Reply:
Done. Extracted these to Darclib.Helpers.FeedConstants